File: scala/ex2_2_4.scala
trait Student(val grade: Float)
trait Teacher(val course: String)

class TA(
    gpa: Float, course: String
) extends Student(gpa) with Teacher(course)

class Volunteer(
    gpa: Float, course: String
) extends Student(gpa) with Teacher(course)

def getGrade(st: Student & Teacher): Float =
  val grade: Float = st.grade
  val course: String = st.course
  0.0

@main def main =
  getGrade(TA(0.0, ""))
  getGrade(Volunteer(0.0, ""))

File: typescript/ex2_2_4.ts
interface Student { grade: number; }
interface Teacher { course: string; }

class TA implements Student, Teacher {
    grade: number = 0;
    course: string = "";
}

class Volunteer implements Student, Teacher {
    grade: number = 0;
    course: string = "";
}

function getGrade(st: Student & Teacher): number {
    let grade: number = st.grade;
    let course: string = st.course;
    return 0;
}

getGrade(new TA());
getGrade(new Volunteer());
